xbmc/messaging/ApplicationMessenger.h

```diff
@@ -400,6 +382,18 @@ class CApplicationMessenger
    */
   void RegisterReceiver(IMessageTarget* target);
 
+  /*!
+   * \brief Set the UI thread id to avoid messenger being dependent on
+   * CApplication to determine if marshaling is required
+   * \param thread The UI thread ID
+   */
+  void SetGUIThread(ThreadIdentifier thread) { m_guiThreadId = thread; }
+
+  /*
+   * \brief Signals the shutdown of the application and message processing
+   */
+  void Shutdown() { m_bStop = true; }
```

Comment:
In most cases we use `Stop` or `Deinit` as method name. It would be nice if we could do the same here.